src/wallet/api/wallet_manager.cpp

```diff
@@ -443,6 +446,28 @@ std::string WalletManagerImpl::resolveOpenAlias(const std::string &address, bool
     return addresses.front();
 }
 
+std::string WalletManager::checkUpdates(const std::string &software, const std::string &subdir, bool testnet)
+{
+#ifdef BUILD_TAG
+    static const char buildtag[] = BOOST_PP_STRINGIZE(BUILD_TAG);
+#else
+    static const char buildtag[] = "source";
+#endif
+
+    std::string version, hash;
+    MDEBUG("Checking for a new " << software << " version for " << buildtag);
+    if (!tools::check_updates(software, buildtag, testnet, version, hash))
```

Comment:
@vtnerd DNSSEC records are signed and verifiable from the root zone down, and that provides the hash for the download. Since the URL is hard-coded there's separation of the available update record and the download. If the DNS records are hijacked the download won't exist or match their hash. Similarly, if the site is hacked the download won't match the hash presented by the DNSSEC-signed record.